qiskit/transpiler/passes/basis/decompose.py

```diff
@@ -11,26 +11,39 @@
 # that they have been altered from the originals.
 
 """Expand a gate in a circuit using its decomposition rules."""
+from typing import Type, Union, List, Optional
+from fnmatch import fnmatch
 
-from typing import Type
-
-from qiskit.circuit.gate import Gate
 from qiskit.transpiler.basepasses import TransformationPass
 from qiskit.dagcircuit.dagcircuit import DAGCircuit
 from qiskit.converters.circuit_to_dag import circuit_to_dag
+from qiskit.circuit.gate import Gate
+from qiskit.utils.deprecation import deprecate_arguments
 
 
 class Decompose(TransformationPass):
     """Expand a gate in a circuit using its decomposition rules."""
 
-    def __init__(self, gate: Type[Gate] = None):
+    @deprecate_arguments({"gate": "gates_to_decompose"})
+    def __init__(
+        self,
+        gate: Optional[Type[Gate]] = None,
+        gates_to_decompose: Optional[Union[Type[Gate], List[Type[Gate]], List[str], str]] = None,
+    ) -> None:
         """Decompose initializer.
 
         Args:
-            gate: gate to decompose.
+            gate: DEPRECATED gate to decompose.
+            gates_to_decompose: optional subset of gates to be decomposed,
+                identified by gate label, name or type. Defaults to all gates.
         """
         super().__init__()
+
+        if not isinstance(gates_to_decompose, list):
+            gates_to_decompose = [gates_to_decompose]
+
         self.gate = gate
```

Comment:
Actually, we probably should also deprecate getting/setting the `gate` attribute, I'm not sure if somewhere in Qiskit this is relied upon. You could do this by adding getters and setters and warn if they are used.